src/Compat/StyleWithNonce.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace ipl\Web\Compat;
+
+use Icinga\Application\Version;
+use Icinga\Util\Csp;
+use ipl\Web\Style;
+
+/**
+ * Use this class to define inline style which is compatible
+ * with Icinga Web &lt; 2.12 and with CSP support in &gt;= 2.12
+ */
+class StyleWithNonce extends Style
+{
+    public function getNonce(): ?string
+    {
+        if ($this->nonce === null) {
+            $this->nonce = version_compare(Version::VERSION, '2.12.0', '>=')
+                ? Csp::getStyleNonce()
+                : '';
```

Comment:
Why empty string instead of null?

Reply:
Because otherwise the version compare and method call would needlessly occur in case of < 2.12

Reply:
..at every getter call